### PR TITLE
Plone 5: provide precompiled bundle.

### PR DESCRIPTION
**Pull request by jone**
- Add basic testing setup (and dev buildout).
- Add uninstall tests.
- Add upgrade step directory.
- Move JS to a resource directory (Plone 4 upgrade step).
- Compile JS for Plone 5 (with upgrade step).

### CODE REVIEW

File: collective/js/throttledebounce/profiles/default_plone5/registry.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<registry>
+
+  <records prefix="plone.bundles/collective-js-throttledebounce-resources"
+           interface="Products.CMFPlone.interfaces.IBundleRegistry">
+    <value key="enabled">True</value>
+    <value key="depends">plone</value>
+    <value key="stub_js_modules">jquery</value>
+    <value key="compile">False</value>
+    <value key="jscompilation">++plone++collective.js.throttledebounce/throttledebounce-compiled.js</value>
```
Comment (busykoala) [inline collective/js/throttledebounce/profiles/default_plone5/registry.xml]: I'm not quite sure but I think this should be `++resource++collective...` according to the directory registry naming.

File: collective/js/throttledebounce/upgrades/20191217181451_provide_compiled_resource_bundles_for_plone_5/registry.xml [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
-      <element>throttledebounce</element>
-    </value>
-
+           interface="Products.CMFPlone.interfaces.IBundleRegistry">
     <value key="enabled">True</value>
-    <value key="depends">plone-legacy</value>
+    <value key="depends">plone</value>
+    <value key="stub_js_modules">jquery</value>
     <value key="compile">False</value>
     <value key="jscompilation">++plone++collective.js.throttledebounce/collective.js.throttledebounce-compiled.js</value>
```
Comment (busykoala) [inline collective/js/throttledebounce/upgrades/20191217181451_provide_compiled_resource_bundles_for_plone_5/registry.xml]: Same here...

**Review by djowett-ftw — APPROVED** [review]
Whizzed through it, but I think that looks good - with the caveat that users outside ftw will have to buy into `ftw.upgrade`, but I think that's not too unreasonable

File: package.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  "version": "1.0.0",
+  "main": "index.js",
+  "repository": "git@github.com:collective/collective.js.throttledebounce.git",
+  "author": "yarn add -D requirejs chokidar-cli",
+  "license": "GPL2",
+  "private": true,
+  "scripts": {
+    "build": "r.js -o build.js",
+    "build-js": "r.js -o build.js",
+    "watch": "chokidar \"./ftw/slider/browser/resources/(*.js|*.css)\" -i \"**/*-compiled.(js|css)\" -c \"if [[ '{path}' = *.js ]]; then r.js -o build.js; fi;\""
```
Comment (busykoala) [inline package.json]: You might want to take notice of @Nachtalb's issue with Posix here and do something like:
`        "watch": "chokidar \"./ftw/slider/browser/resources/(*.js|*.css)\" -i \"**/*-compiled.(js|css)\" -c \"/bin/sh -c \\\"if [[ '{path}' = *.js ]]; then r.js -o build.js; else r.js -o buildcss.js; fi;\\\"\""
` (except of the css part and therefore condition).

File: collective/js/throttledebounce/upgrades/20191217181451_provide_compiled_resource_bundles_for_plone_5/registry.xml (hunk 2 above)
Comment (djowett-ftw) [reply collective/js/throttledebounce/upgrades/20191217181451_provide_compiled_resource_bundles_for_plone_5/registry.xml]: Oh - yes - that's true.  Should stop some TTW overwriting issues we don't want

File: collective/js/throttledebounce/profiles/default_plone5/registry.xml (hunk 1 above)
Comment (jone) [reply collective/js/throttledebounce/profiles/default_plone5/registry.xml]: 👍 thanks

File: collective/js/throttledebounce/upgrades/20191217181451_provide_compiled_resource_bundles_for_plone_5/registry.xml (hunk 2 above)
Comment (jone) [reply collective/js/throttledebounce/upgrades/20191217181451_provide_compiled_resource_bundles_for_plone_5/registry.xml]: 👍 thanks

File: package.json (hunk 3 above)
Comment (jone) [reply package.json]: 👍 thanks

**Comment by jone** [comment]
Updated

File: package.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  "version": "1.0.0",
+  "main": "index.js",
+  "repository": "git@github.com:collective/collective.js.throttledebounce.git",
+  "author": "yarn add -D requirejs chokidar-cli",
+  "license": "GPL2",
+  "private": true,
+  "scripts": {
+    "build": "r.js -o build.js",
+    "build-js": "r.js -o build.js",
+    "watch": "chokidar \"./ftw/slider/browser/resources/(*.js|*.css)\" -i \"**/*-compiled.(js|css)\" -c \"/bin/sh -c \\\"if [[ '{path}' = *.js ]]; then r.js -o build.js; else r.js -o buildcss.js; fi;\\\"\""
```
Comment (Nachtalb) [inline package.json:12]: You could replace the command with a simple `yarn build` and remove the watch of `*.css` files.

**Review by Nachtalb — APPROVED** [review]

File: package.json (hunk 4 above)
Comment (jone) [reply package.json:12]: hmm, I think it should work now and it's like we do it normally, so I'd stick with it 😉

**Pull request merged**